File: src/main/kotlin/org/kite9/diagram/common/elements/AbstractTemporaryConnected.java
package org.kite9.diagram.common.elements;

import org.kite9.diagram.common.elements.factory.AbstractDiagramElement;
import org.kite9.diagram.common.elements.factory.TemporaryConnected;
import org.kite9.diagram.model.*;

import java.util.ArrayList;
import java.util.Collection;

public abstract class AbstractTemporaryConnected extends AbstractDiagramElement implements TemporaryConnected {

	protected String id;
	private Collection<Connection> links = new ArrayList<>();

	public AbstractTemporaryConnected(DiagramElement parent) {
		super(parent);
	}

	@Override
	public String getID() {
		return id;
	}

	public AbstractTemporaryConnected() {
		super();
	}

	@Override
	public Collection<Connection> getLinks() {
		return links;
	}

	public Connection getConnectionTo(Connected c) {
		for (Connection link : getLinks()) {
			if (link.meets(c)) {
				return link;
			}
		}
	
		return null;
	}

	public boolean isConnectedDirectlyTo(Connected c) {
		return getConnectionTo(c) != null;
	}

	@Override
	public Container getContainer() {
		return (Container) getParent();
	}

}

File: src/main/kotlin/org/kite9/diagram/common/elements/factory/AbstractTemporaryConnected.kt
package org.kite9.diagram.common.elements.factory

import org.kite9.diagram.model.Connected
import org.kite9.diagram.model.Connection
import org.kite9.diagram.model.Container
import org.kite9.diagram.model.DiagramElement

abstract class AbstractTemporaryConnected(private val _id: String, private val p: DiagramElement) : AbstractDiagramElement(p), TemporaryConnected {

    private val links: Collection<Connection> = ArrayList()

    override fun getLinks(): Collection<Connection> {
        return links
    }

    override fun getConnectionTo(c: Connected): Connection? {
        for (link in getLinks()) {
            if (link.meets(c)) {
                return link
            }
        }
        return null
    }

    override fun isConnectedDirectlyTo(c: Connected): Boolean {
        return getConnectionTo(c) != null
    }

    override fun getContainer(): Container {
        return getParent() as Container
    }

    override fun getID(): String {
        return _id
    }
}